pkg/controller/mpi_job_controller.go

```diff
@@ -585,9 +577,48 @@ func (c *MPIJobController) syncHandler(key string) error {
 		return err
 	}
 
+	if launcher != nil {
+		launcherSuspendUpdate := false
+		if isMPIJobSuspended(mpiJob) && !isJobSuspended(launcher) {
```

Comment:
You could have a single condition:
```suggestion
		if isMPIJobSuspended(mpiJob) != isJobSuspended(launcher) {
```

Reply:
Done